source/plugin/economy/practices.rst

```diff
@@ -59,8 +61,8 @@ Here's how you **should** withdraw money:
     Account account = ...;
     BigDecimal requiredAmount = BigDecimal.valueOf(20);
 
-    TransactionResult result = account.withdraw(service.getDefaultCurrency(),
-        requiredAmount, Cause.source(this).build());
+    TransactionResult result = account.withdraw(service.getDefaultCurrency(), requiredAmount,
+                Cause.of(EventContext.of(ImmutableMap.of(EventContextKeys.PLUGIN, this)), this));
```

Comment:
It would be nice if I could get some feedback whether this is the expected way to specify the cause in this case. Or should more things added to the context/cause?

Reply:
You should be using the `PluginContainer`. 

As for the context, the alternative is 

`Cause.of(EventContext.builder().add(EventContextKeys.PLUGIN, container).build(), container);`

but that's just a matter of deciding which one is clearer. 

Sidenote: keep in mind that the builder is using an `HashMap` and the `EventContext` is using an `ImmutableMap`, so if you decide to use the builder you're iterating twice over the map.

See https://github.com/SpongePowered/SpongeAPI/blob/4f54ce0d8e487d139d3dcf9e39b29903c89b514b/src/main/java/org/spongepowered/api/event/cause/EventContext.java#L85 and https://github.com/SpongePowered/SpongeAPI/blob/4f54ce0d8e487d139d3dcf9e39b29903c89b514b/src/main/java/org/spongepowered/api/event/cause/EventContext.java#L224